widgetry/src/widgets/button.rs

```diff
@@ -195,8 +195,64 @@ impl Btn {
         }
     }
 
-    pub fn custom(normal: GeomBatch, hovered: GeomBatch, hitbox: Polygon) -> BtnBuilder {
-        BtnBuilder::Custom(normal, hovered, hitbox, None)
+    pub fn pop_up<I: Into<String>>(ctx: &EventCtx, label: Option<I>) -> BtnBuilder {
```

Comment:
Nice, the overall approach of this is quite elegant! The caller doesn't care how it happens, and the finer control over layout is easy to understand.